docs/protocol.md

```diff
@@ -0,0 +1,364 @@
+            },
+            "prompt_index": 0
+        }
+    ],
+    "system_fingerprint": null,
+    "usage": {
+        "completion_tokens": 64,
+        "prompt_tokens": 967,
+        "total_tokens": 1031
+    }
+}
+```
+
+### Streaming response
+
+This response is based off the [OpenAI chat completion chunk object](https://platform.openai.com/docs/api-reference/chat/streaming),
+with additional properties needed to display sources and citations properly.
+
+The first chunk contains a choice with the `context` property, since that is available before the answer, and subsequent chunks contain the answer to the question.
+
+* `"choices"`: A list of responses from the LLM, typically containing only 1 response as our app sets `n=1` when requesting a completion. Each response contains:
+   * `"delta"`: An object containing the actual content of the response. Copied directly from the OpenAI response.
+   * `"finish_reason"`: A string representing the finish state of the response. Copied directly from the OpenAI response.
+   * `"index"`: A number indicating which response this is (0 in the case of 1 response given). Copied directly from the OpenAI response.
+   * `"content_filter_results"`: An object from the Azure Content Safety filter. Copied directly from the OpenAI response, but will only be returned when using Azure OpenAI, not openai.com OpenAI.
+   * `"context"`: An object containing additional details needed for the RAG chat app. Includes:
+       * `"data_points"`: An object containing text and/or image data chunks, a list in the `"text"` or `"images"` properties. Useful for citations.
+       * `"thoughts"`: A list describing each step of the RAG approach, useful for a debug display in the RAG chat app.
+   * `"session_state"`: An object containing the "memory" for the chat app, such as a user ID. Not currently used by this app.
+* `"created"`: The Unix timestamp (in seconds) of when the chat completion was created. Copied directly from the OpenAI response.
+* `"id"`: A unique identifier for the chat completion. Copied directly from the OpenAI response.
+* `"model"`: The model used for the completion, such as "gpt-35-turbo". Copied directly from the OpenAI response.
+* `"object"`: The object type, always "chat.completion". Copied directly from the OpenAI response.
+* `"prompt_filter_results"`: Copied directly from the OpenAI response, but will only be returned when using Azure OpenAI, not openai.com OpenAI.
+* `"system_fingerprint"`: Represents the backend configuration that the model runs with. Copied directly from the OpenAI response.
+* `"usage"`: Usage statistics for the completion request. Copied directly from the OpenAI response.
+
+
+Here's an example of the first two JSON objects streamed for a `/chat` request:
+
+```json
+{
+    "choices": [
+        {
+            "delta": {
+                "role": "assistant"
+            },
+            "context": {
+                "data_points": {
+                    "text": [
```

Comment:
is there a gpt4/vision response in the doc

Reply:
I added an images example here:

https://github.com/Azure-Samples/ai-chat-app-protocol/pull/1/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R532